File: GXpert/GXpert.Web/Modules/ServerTypes/Users/TeacherForm.ts
import { StringEditor, LookupEditor, DateEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface TeacherForm {
    Prn: StringEditor;
    Name: StringEditor;
    Email: StringEditor;
    Mobile: StringEditor;
    UserId: LookupEditor;
    StateId: LookupEditor;
    DistrictId: LookupEditor;
    TalukaId: LookupEditor;
    Dob: DateEditor;
    SchoolId: LookupEditor;
    Description: TextAreaEditor;
}

export class TeacherForm extends PrefixedContext {
    static readonly formKey = 'Users.Teacher';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TeacherForm.init)  {
            TeacherForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = DateEditor;
            var w3 = TextAreaEditor;

            initFormType(TeacherForm, [
                'Prn', w0,
                'Name', w0,
                'Email', w0,
                'Mobile', w0,
                'UserId', w1,
                'StateId', w1,
                'DistrictId', w1,
                'TalukaId', w1,
                'Dob', w2,
                'SchoolId', w1,
                'Description', w3
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Users/Teacher/TeacherForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Users.Forms;

[FormScript("Users.Teacher")]
[BasedOnRow(typeof(TeacherRow), CheckNames = true)]
public class TeacherForm
{
    [HalfWidth]
    public string Prn { get; set; }
    [HalfWidth]
    public string Name { get; set; }
    [HalfWidth]
    public string Email { get; set; }
    [HalfWidth]
    public string Mobile { get; set; }
    [HalfWidth]
    public int UserId { get; set; }
    
    [HalfWidth]
    public int StateId { get; set; }
    [HalfWidth]
    public int DistrictId { get; set; }
    [HalfWidth]
    public int TalukaId { get; set; }
    [HalfWidth]
    public DateTime Dob { get; set; }
    [HalfWidth]
    public int SchoolId { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
}